### ISSUE TITLE
DefaultFrameBuffer::new is unsound with arbitrary global allocators

### ISSUE THREAD

**Issue by nox** [rust-av/rust-av#41]
```rust
let size = video.size(ALIGNMENT);
let data = unsafe {
    Heap.alloc(Layout::from_size_align(size, ALIGNMENT).unwrap())
        .unwrap()
};
let buf = BytesMut::from(unsafe { Vec::from_raw_parts(data, size, size) });
```

`ALIGNMENT ` is 32, whereas the default alignment for `Vec<T>` is generally lower. The allocation stuff in Rust pass the type alignment to the freeing method, but this code doesn't implement `Drop` to free the allocated vector by hand with the proper alignment.

**Comment by lu-zero** [comment]
You are correct.

**Comment by lu-zero** [comment]
A simple solution sort of depends on https://github.com/carllerche/bytes/issues/86 being available.

**Comment by Luni-4** [comment]
This issue should be fixed by d106d52ff7004c1fba04023351126b0d7f087ebd
